addons/website/views/snippets/s_social_media.xml

```diff
@@ -2,7 +2,7 @@
 <odoo>
 
 <template id="s_social_media" name="Social Media">
-    <div class="s_social_media text-left o_not_editable s_preview" contenteditable="false">
```

Comment:
Is it actually useless? We have fixes incoming regarding this but I think right now, it's still useful?

Reply:
For snippets that are drag and dropped it will be added by the editor (see https://discord.com/channels/@me/917694309015384137/956155765960343552). The incoming fixes are for snippets already on the page.

Reply:
We don't have access to this link

Reply:
Since this line is outdated, I am not too sure what it was about, but I guess you are speaking about the `contenteditable` stuff?
I don't think any bugfix is related to this, as it is added on purpose, to have a quick and easy way to disable the edition of the whole snippet except the title where we set it to true.
Explanation is here https://github.com/odoo/odoo/pull/80217#discussion_r787612278

Sorry if I am off topic :face_with_spiral_eyes:

Reply:
I updated the comment

Reply:
Yes, so the point of my comment is that the `contenteditable=false` is not actually necessary, because the editor will add it based on the `o_not_editable` class (it will prepatch the snippets in the snippets menu that are previously loaded from the xml). So once you drop a snippet, it (currently) won't make any difference.

For example in commit 30db617bc8ff7727f40d7ef58c6578e84a13f284 (taken from the comment mentioned), the same logic applies. The fix only depends on the `o_not_editable` class and the `contenteditable=true` attribute.

Of course, in light of the fixes being worked on, it might be useful to discuss what _should_ actually be included in the xml. For example, in both cases, there is no equivalent class that will result in adding the `contenteditable=true` attribute... (a class named `o_editable` does exist, but it does not actually work the same way)

Reply:
> Yes, so the point of my comment is that the contenteditable=false is not actually necessary,

It is mandatory, if you remove the `contenteditable=false` on the parent, you can then add text in between the icons.
Not only that, but you can even create links in between, messing with the options in the right panel which depends of immediate child links, which will just raise TB at the end. (And which is not worth fixing as we don't want to introduce the root cause in the first place, nothing should be possible to add between icons, just adding stuff in the title is allowed)

That's exactly what I listed here https://github.com/odoo/odoo/pull/80217#discussion_r787612278

![image](https://user-images.githubusercontent.com/30048408/159916659-b89ec91b-053a-422d-bf98-ade24e178a2d.png)

Reply:
Where did you test this? On the latest runbot I'm unable to reproduce this... When inspecting the DOM the `contenteditable=false` is there, even if it is not in the xml.

![image](https://user-images.githubusercontent.com/7951286/159919632-136943c8-817f-499d-85b3-6aa30e72456a.png)

Reply:
Latest 15.2

Reply:
> so I don't understand how you could have tested this

:thinking: :thinking: by opening the HTML editor and writing whatever I want inside it to test it? Copy paste the snippet without the attribute, everything fails.

Reply:
Here is a A to Z video about it, from runbot to the traceback with all bugs included: https://watch.screencastify.com/v/VusiwK40x8aPYF6NCv2y

Note that you could also very well do the same by going in the backend > technical > views (in debug) and edit the social media snippet there to remove the attribute. Plenty of ways.
Also you could juste drag & drop it from the builder, then remove it through the html editor, or in the backend.

Reply:
Ok so, that's exactly what you shouldn't do :slightly_smiling_face: It completely misses the point of my comment... Because even if you remove the attribute from the xml it will still be there in the snippet after you drag and drop.

Reply:
That certainly seems like a weak solution to remove it from the snippet definition, because the editor will add it later on drag & drop for multiple reason:
1. People will copy paste this template "in place" elsewhere, without the attribute
2. Snippet can be used with `t-snippet-call` not going through the drag & drop phase, even if that case seems a bit "meeeh" and not an usual flow.

Reply:
No, you're misunderstanding, the editor does not actually add the attribute in the drag and drop phase... It patches them in the snippets menu, so that when new snippets are created by dragging and dropping them in the page the attribute will be there. This is also why the related bugs are only present on saved snippets, since the `contenteditable` attributes will be removed when saving and won't be added back when editing again.

In the first case, generally speaking, if the attribute isn't present on the original template, it shouldn't be on the copied template either (the `o_not_editable` class will be copied of course, in this particular case). So this actually leads us back to a discussion of what _should_ be included but I'm pretty sure that this discussion shouldn't be held here (since work is being done on fixing related bugs). But anyways: of course you can't leave `contenteditable=true` attributes in a saved DOM, so it seems only logical that the editor would take care of all `contenteditable` attributes (based on for example an `o_not_editable` class that people will then copy paste wherever they want).

The second case doesn't apply, because `t-snippet-call` inclusions are resolved even before the patching of the snippets in the snippets menu.

Reply:
> No, you're misunderstanding, the editor does not actually add the attribute in the drag and drop phase... It patches them in the snippets menu, so that when new snippets are created by dragging and dropping them in the page the attribute will be there.

By drag & dropping phase, understand adding a snippet in the DOM through the right panel, not the exact moment you click on it to drag & drop it ^^
Just opposed to copying the snippet manually in the DOM through the HTML editor or any other way not going through the right panel :-).

Exactly as you said it here (in a probably better way):
> Because even if you remove the attribute from the xml it will still be there in the snippet **after you drag and drop.**

> So this actually leads us back to a discussion of what should be included but I'm pretty sure that this discussion shouldn't be held here (since work is being done on fixing related bugs)

It has to be decided here, since that's here that you want to remove the `contenteditable=false` from the snippet definition.
The fix is not yet shipped so we shouldn't rely on it and ship this one without it.

Anyway, I was missing something: the contenteditable=false will prevent the TB/crashes/broken snippet **only** during the first edition, as leaving the edit mode will remove it anyway.
So my argument is a bit less pertinent.

Also, other solution would be that this PR should await for the fix before getting shipped.
(In that case, the `contenteditable=true` should remain as we don't have a class to add it AFAIK)

Reply:
Ok, final comment :slightly_smiling_face: Not trying to annoy you guys, but I feel there are still some misconceptions floating around. If you prefer to leave it in untouched in the PR, that's totally fine, I just wanted to make sure you're getting the right information.

About the comment by @rdeodoo:

> Just opposed to copying the snippet manually in the DOM through the HTML editor or any other way not going through the right panel :-).
>
> Exactly as you said it here (in a probably better way):
>
> > Because even if you remove the attribute from the xml it will still be there in the snippet after you drag and drop.

Missing the point of my original comment here... This is not the same. Editing it in the xml (the actual xml file) won't affect the html in the DOM. THe xml can only ever affect the first edition because the second edition is based on what is already in the DOM.

> It has to be decided here, since that's here that you want to remove the contenteditable=false from the snippet definition.
> The fix is not yet shipped so we shouldn't rely on it and ship this one without it.
Removing the attribute does not rely on the fix.

> Anyway, I was missing something: the contenteditable=false will prevent the TB/crashes/broken snippet only during the first edition, as leaving the edit mode will remove it anyway.
> So my argument is a bit less pertinent.

It actually makes no difference at all, not in the first or the second edition, since removing it does not actually affect the html being dropped.

> (In that case, the contenteditable=true should remain as we don't have a class to add it AFAIK)

That is true, but at the same time problematic. Not in the scope of this PR however. Since we can't leave it in the DOM after save (resulting in a weirdly editable page), it really should have a class (or use the existing `o_editable`) so we can decide where to add them back when reentering edit mode.

About the comment by @qsm-odoo:

> * true, the contenteditable attribute is useless in the snippet definition even with the bugs, for the first edition (after d&d)

This also holds for the second edition. The second edition is never affected by the original xml, only by the saved DOM.

> * true, the contenteditable attribute is not useless in the snippet definition, as it allows to copy/paste it/t-call it the right way considering the current bugs (but only works for the first edition too anyway because of the bugs)

It doesn't affect the `t-call` flow, as for the copy/pasting, the contents of the paste depend on where you are copying it from:

* Copying from the xml file directly seems weird
* Any other source will not be affected by the contents in the xml file

But even in the first case, the _right way_ depends of course where you are pasting it:

* Pasting in another snippet definition: the same logic applies
* Pasting in the saved html of a page: the paste should probably reflect the html after an eventual `cleanForSave` which does not include the attribute
* Pasting inside html being edited: a weird thing to do, since you can just drag and drop the snippet

Reply:
Final comment too ;-)

You are missing the point, we are not talking about copy paste of end user but of developers and community:
- Another team copying the snippet in their refactor of the product page eg
- A community module or theme or whatever copying the snippet in their template

That's a common flow, as we do it ourselves at many places in our code base.

tl;dr:
- Removing it will make the bug leak elsewhere when copied
- Keeping it will prevent that, but won't be the ultimate solution as it only helps for the first edition (which will end up with the case of the video I already mentioned https://watch.screencastify.com/v/VusiwK40x8aPYF6NCv2y)

Since there is no benefit whatsoever to remove it but keeping it prevents an issue (small one I agree), at least until this is fixed, it seems better to keep it.